admin/abstract/views.py

```diff
@@ -0,0 +1,49 @@
+from django.views.generic import FormView
+from django.http import HttpResponseNotFound
+from django.shortcuts import render
+
+
+class GuidFormView(FormView):
+    form_class = None
+    template_name = None
+    object_type = None
+
+    def get_guid_object(self):
+        raise NotImplementedError
+
+    def __init__(self):
+        self.guid = None
+        super(GuidFormView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        self.guid = request.GET.get('guid', None)
+        if self.guid is not None:
+            try:
+                guid_object = self.get_guid_object()
+            except (AttributeError, TypeError):
+                return HttpResponseNotFound(
```

Comment:
This could be a bit more user friendly :smiling_imp:

https://docs.djangoproject.com/en/1.9/ref/urls/#handler404 Easy way to create a global django 404 handler.